common/djangoapps/third_party_auth/views.py

```diff
@@ -28,6 +30,8 @@ def inactive_user_view(request):
     # 'next' may be set to '/account/finish_auth/.../' if this user needs to be auto-enrolled
     # in a course. Otherwise, just redirect them to the dashboard, which displays a message
     # about activating their account.
+    profile = UserProfile.objects.get(user=request.user)
+    compose_and_send_activation_email(request.user, profile)
```

Comment:
We are using `request.user` here, means the session has been set already. 
Where we are setting the session for this new user?